tests/reftest/mod.rs

```diff
@@ -49,36 +49,55 @@ fn reftest_decoder<T: std::io::Read>(mut decoder: jpeg::Decoder<T>, path: &Path,
     }
 
     let ref_file = File::open(ref_path).unwrap();
-    let (ref_info, mut ref_reader) = png::Decoder::new(ref_file).read_info().expect("png failed to read info");
+    let mut decoder = png::Decoder::new(ref_file);
+
+    if pixel_format == jpeg::PixelFormat::L16 {
+        // disable the default 8bit output of png v0.16.8 (fixed in master branch of png)
+        decoder.set_transformations(png::Transformations::EXPAND);
+    }
+    
+    let (ref_info, mut ref_reader) = decoder.read_info().expect("png failed to read info");
 
     assert_eq!(ref_info.width, info.width as u32);
     assert_eq!(ref_info.height, info.height as u32);
-    assert_eq!(ref_info.bit_depth, png::BitDepth::Eight);
 
     let mut ref_data = vec![0; ref_info.buffer_size()];
     ref_reader.next_frame(&mut ref_data).expect("png decode failed");
     let mut ref_pixel_format = ref_info.color_type;
 
-    if ref_pixel_format == png::ColorType::RGBA {
+    if ref_pixel_format == png::ColorType::RGBA { 
         ref_data = rgba_to_rgb(&ref_data);
         ref_pixel_format = png::ColorType::RGB;
     }
+    let mut refdata_16 : Vec<u16> = ref_data.iter().map(|x| *x as u16).collect();
+    let mut data_u16 : Vec<u16> = data.iter().map(|x| *x as u16).collect();
```

Comment:
If you move the initial assignment to the two branches `L8` and `RGB24`—accepting that this will duplicate it—Rust will happily allow you to remove the initial assignment and `mut` qualifier here.

Reply:
Not sure if have realized Rust can do this but you can declare the variables without initialization. As long as you initialize them in each branch of the match the compiler will happily treat them as initialized afterwards.

```rust
let refdata_16: Vec<u16>;
let data_u16: Vec<u16>;

match pixel_format {
    L8 => {
        refdata_16 = …;
        data_u16 = …;
    }
    …
}
```

No need to change the code another time but sometimes this is clearer.